### PR TITLE
[docs][Ray Clusters] New Ray Clusters getting started page.

### LINKED ISSUE #27130: [Ray clusters][docs] Write the new Ray Clusters: Getting Started page
The top-level Getting Started page should direct users, based on what they are trying to accomplish, to the correct page between "Managing Ray Clusters on Kubernetes", "Managing Ray Clusters via VMs", and "Using Ray Clusters". There should be some discussion on Kubernetes vs. VMs for those looking to create new Ray clusters for the first time.

### PR DESCRIPTION
**Pull request by cadedaniel**
Signed-off-by: Cade Daniel <cade@anyscale.com>

## Why are these changes needed?
This PR introduces a new Ray Clusters Getting Started page. It will serve as a top-level landing page for new users, and will direct them to the Ray Cluster Launcher (for VMs) or KubeRay getting started pages.

## Related issue number

Closes #27130

### CODE REVIEW

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 1]
```diff
@@ -26,13 +33,25 @@ One of Ray's strengths is the ability to leverage multiple machines for
...
 done so often), but the real power is using Ray on a cluster of machines.
 
-Ray can automatically interact with the cloud provider to request or release
-instances. You can specify :ref:`a configuration <cluster-config>` to launch
-clusters on :ref:`AWS, GCP, Azure (community-maintained), Aliyun (community-maintained), on-premise, or even on
-your custom node provider <cluster-cloud>`. Ray can also be run on :ref:`Kubernetes <kuberay-index>` infrastructure.
-Your cluster can have a fixed size
-or :ref:`automatically scale up and down<cluster-autoscaler>` depending on the
-demands of your application.
+A Ray Cluster can be created from virtual machines using the :ref:`Ray Cluster Launcher<ref-cluster-quick-start-vms-under-construction>`, which is officially supported on AWS and GCP and community supported on Azure and Aliyun, or created from machines managed by Kubernetes via :ref:`KubeRay<kuberay-index>`, which is the officially supported way to deploy Ray on Kubernetes.
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:36]: This sentence is long. It should be broken up.

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 2]
```diff
@@ -26,13 +33,25 @@ One of Ray's strengths is the ability to leverage multiple machines for
...
+Ray can also be :ref:`deployed on-premise<cluster-private-setup-under-construction>` or even onto infrastructure not listed here by :ref:`providing a custom node provider<additional-cloud-providers-under-construction>`.
+
+Your cluster can have a fixed size, or :ref:`leverage autoscaling<cluster-autoscaler-under-construction>` to automatically request additional or release unused nodes depending on the demands of your application.
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:39]: Nit/suggestion: `to automatically request additional resources or release unused nodes, according to the demands of your application.`

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 3]
```diff
@@ -42,56 +61,44 @@ Where to go from here?
     :column: col-lg-6 px-2 py-2
     :card:
 
-    **Quick Start** 
+    **Getting Started with the Ray Clusters on Virtual Machines**
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:64]: there's a spare `the`

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 4]
```diff
@@ -42,56 +61,44 @@ Where to go from here?
...
     ---
 
-    **Deployment Guide**
+    **Getting Started with Ray on Kubernetes**
     ^^^
 
-    Learn how to set up a distributed Ray cluster and run your workloads on it.
+    In this quick start tutorial you will take a sample application designed to
+    run on a laptop and scale it up in the cloud, using a cloud provider such as
+    AWS or GCP.
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:83]: Suggestion based on the content of that quickstart:
```
You will learn how to deploy a Ray application to a Kubernetes cluster.
You can run the tutorial on a remote Kubernetes cluster or on your laptop via KinD.
```

**Comment by DmitriGekhtman** [comment]
looks pretty good so far, left some comments on wording

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 5]
```diff
@@ -42,56 +35,39 @@ Where to go from here?
...
-
-    **API**
-    ^^^
-
-    Get more in-depth information about the various APIs to interact with Ray
-    Clusters, including the :ref:`Ray cluster config YAML and CLI<cluster-config>`,
-    the :ref:`Ray Client API<ray-client>` and the
-    :ref:`Ray job submission API<ray-job-submission-api-ref>`.
+    Understand the key concepts behind Ray Clusters. Learn about the main
+    concepts and the different ways to interact with a cluster.
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:65]: This description could be made less redundant -- for example, the word "concepts" should appear just once.

**Review by DmitriGekhtman — APPROVED** [review]
Nit on wording in Key Concepts description.
Overall, nice! Looks concise and efficient.

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 6]
```diff
@@ -26,13 +15,17 @@ One of Ray's strengths is the ability to leverage multiple machines for
 distributed execution. Ray can, of course, be run on a single machine (and is
 done so often), but the real power is using Ray on a cluster of machines.
```
Comment (richardliaw) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:16]:
```suggestion
distributed execution. Ray can be run on a single machine and is
done so often, but the real power of Ray is the ability to seamlessly scale to a cluster of machines.
```

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 7]
```diff
@@ -26,13 +15,17 @@ One of Ray's strengths is the ability to leverage multiple machines for
...
 
-Ray can automatically interact with the cloud provider to request or release
-instances. You can specify :ref:`a configuration <cluster-config>` to launch
-clusters on :ref:`AWS, GCP, Azure (community-maintained), Aliyun (community-maintained), on-premise, or even on
-your custom node provider <cluster-cloud>`. Ray can also be run on :ref:`Kubernetes <kuberay-index>` infrastructure.
-Your cluster can have a fixed size
-or :ref:`automatically scale up and down<cluster-autoscaler>` depending on the
-demands of your application.
+A Ray cluster is a set of one or more nodes that are running Ray and share the same :ref:`head node<cluster-head-node-under-construction>`.
+Ray clusters can be a fixed-size number of nodes, or :ref:`autoscale<cluster-autoscaler-under-construction>` the cluster size up and down according to the demand of the Ray workload.
```
Comment (richardliaw) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:19]:
```suggestion
Ray clusters can either be a fixed-size number of nodes or :ref:`can autoscale<cluster-autoscaler-under-construction>` (i.e., automatically provision or deprovision the number of nodes in a cluster) according to the demand of the Ray workload.
```

File: doc/source/cluster/cluster_under_construction/getting-started.rst (hunk 5 above)
Comment (richardliaw) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:65]:
```suggestion
    Understand the key concepts and main ways of interacting with a Ray cluster.
```

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 8]
```diff
@@ -42,56 +35,39 @@ Where to go from here?
...
-    .. link-button:: cluster-key-concepts
+    .. link-button:: kuberay-quickstart
         :type: ref
-        :text: Learn Key Concepts
+        :text: Getting Started with Ray on Kubernetes
         :classes: btn-outline-info btn-block
     ---
 
-    **Deployment Guide**
+    **I want to learn key Ray concepts**
```
Comment (richardliaw) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:62]: should this be first on the list?

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 9]
```diff
@@ -42,56 +35,39 @@ Where to go from here?
     :column: col-lg-6 px-2 py-2
```
Comment (richardliaw) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:35]: can we do px-3 here?

File: doc/source/cluster/cluster_under_construction/getting-started.rst (hunk 6 above)
Comment (DmitriGekhtman) [reply doc/source/cluster/cluster_under_construction/getting-started.rst:16]: This is still ever-so-slightly ungrammatical.
How about:
`Ray is great for multiprocessing on a single machine. However, the real power of Ray is the ability to seamlessly scale to a cluster of machines.`

Comment (richardliaw) [reply doc/source/cluster/cluster_under_construction/getting-started.rst:16]: good catch... didn't read that paren phrase well

File: doc/source/cluster/cluster_under_construction/getting-started.rst (hunk 5 above)
Comment (DmitriGekhtman) [reply doc/source/cluster/cluster_under_construction/getting-started.rst:65]: (Covered by Richard's suggestion below.)

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 10]
```diff
@@ -23,75 +12,63 @@ What is a Ray cluster?
...
-
-    +++
-    .. link-button:: ref-cluster-quick-start-vms-under-construction
-        :type: ref
-        :text: Ray Clusters Quick Start
-        :classes: btn-outline-info btn-block
-    ---
-
-    **Key Concepts**
+    **I want to learn key Ray concepts**
```
Comment (richardliaw) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:39]:
```suggestion
    **I want to learn key Ray cluster concepts**
```

File: doc/source/cluster/cluster_under_construction/getting-started.rst [hunk 11]
```diff
@@ -23,75 +12,63 @@ What is a Ray cluster?
 ----------------------
 
 One of Ray's strengths is the ability to leverage multiple machines for
-distributed execution. Ray can, of course, be run on a single machine (and is
-done so often), but the real power is using Ray on a cluster of machines.
+distributed execution. Ray is great for multiprocessing on a single machine.
+However, the real power of Ray is the ability to seamlessly scale to a cluster
+of machines
```
Comment (richardliaw) [inline doc/source/cluster/cluster_under_construction/getting-started.rst:17]:
```suggestion
of machines.
```

**Review by richardliaw — APPROVED** [review]

**Pull request merged**